### PR TITLE
FIX: post mover validation color and message

### PR DESCRIPTION
**Pull request by lis2**
When the record is not saved, we should display a proper message.
One potential reason can be plugins for example discourse-calendar is specifying that only first post can contain event.

Before
Screen Shot 2022-01-24 at 2 46 31 pm

After
Screen Shot 2022-01-24 at 2 36 40 pm

### CODE REVIEW

**Comment by tgxworld** [comment]
Can we add a request spec for this? :)

**Comment by martin-brennan** [comment]
Request failed successfully!

File: spec/requests/topics_controller_spec.rb [hunk 1]
```diff
@@ -326,6 +326,19 @@
...
           expect(result['url']).to be_blank
         end
+
+        it 'returns validation error' do
+          PostCreator.any_instance.stubs(:skip_validations?).returns(false)
+          p1.update_columns(raw: "i", cooked: "")
+          post "/t/#{topic.id}/move-posts.json", params: {
+            post_ids: [p1.id],
+            destination_topic_id: dest_topic.id
+          }
```
Comment (tgxworld) [inline spec/requests/topics_controller_spec.rb:336]:
```suggestion
          PostCreator.any_instance.stubs(:skip_validations?).returns(false)
          p1.update_columns(raw: "i", cooked: "")
          post "/t/#{topic.id}/move-posts.json", params: {
            post_ids: [p1.id],
            title: 'a'
          }
```

I feel like the stub doesn't reflect a real world problem. How about using the `title` attribute instead to trigger the `ActiveRecord::NotSaved` error.

Comment (lis2) [reply spec/requests/topics_controller_spec.rb:336]: It was a good idea. I checked it and this is not going to work. Move to new title is calling create! Which triggers InvalidRecord https://github.com/discourse/discourse/blob/main/app/models/post_mover.rb#L42

When in this scenario, we want to trigger this https://github.com/discourse/discourse/blob/main/lib/post_creator.rb#L257

How this validation was even triggered by discourse-calendar why we have skip_validation=true? The answer is that core is ignoring validations. I guess that assumption is that valid post before is valid post after:

https://github.com/discourse/discourse/blob/main/app/models/post.rb#L65

Plugin is registering new validator which is not checking for skip_validation. Behaviour is inconsistent but also correct, as moving a post with event to an existing topic is not making that post valid anymore

https://github.com/discourse/discourse-calendar/blob/main/plugin.rb#L369

That was the reason for my stub, to somehow fake NotSaved

Question is if  in general, when we move posts, if we should skip validations

**Review by tgxworld — APPROVED** [review]

**Comment by tgxworld** [comment]
@lis2 Btw please tag this 2.8 or 2.9 accordingly. It feels like a small change but I'm not sure if we want to risk it so close to the release.

**Pull request merged**